plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/CountingAccessFileHiveMetastore.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.PartitionStatistics;
+import io.trino.plugin.hive.authentication.HiveIdentity;
+import io.trino.plugin.hive.metastore.Database;
+import io.trino.plugin.hive.metastore.MetastoreConfig;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CountingAccessFileHiveMetastore
```

Comment:
It looks like we are applying `io.trino.plugin.hive.metastore.thrift.ThriftMetastoreStats` in the wrong place, so we collect stats only when thrift metastore is used. However collecting statistics could be also useful here for iceberg or for glue. If we could move it out and apply stats collection at different layer then `CountingAccessFileHiveMetastore` would not be needed.

It looks like it is in line with https://github.com/trinodb/trino/pull/8618/files#r674686854

Reply:
it is possible to reuse, but it would still require

- as you observed, applying ThriftMetastoreStats on a different level; but then it would impact JMX names, so it could be a breaking change for monitoring tools 
- adapter code here in the test, since ThriftMetastoreStats does collect much more than what we need, and in a structure optimized for JMX, not for tests

I wouldn't want to reuse `ThriftMetastoreStats` here.